File: cos/olympus/src/main/java/cos/olympus/game/SpellStrategy.java
package cos.olympus.game;

import cos.olympus.game.events.Damage;
import cos.olympus.game.events.Spell;

import java.util.Collection;

public interface SpellStrategy {
    //        val action: SpellAction
    int id();
//        fun inZone(creature: Creature): Boolean
//        fun handle(time: Tsm, actions: ActionConsumer, map: GameMap): Boolean


    boolean onTick(int tick, Collection<Damage> damages);

    boolean inZone(Creature cr);
    boolean isFinished();
    Spell spell();
}

File: cos/cos.olympus/src/main/kotlin/cos/olympus/game/spells/SpellStrategy.kt
//package cos.olympus.game.spells
//
//import cos.olympus.Tsm
//import cos.olympus.game.ActionConsumer
//import cos.olympus.game.Creature
//import cos.olympus.game.GameMap
//import cos.olympus.game.actions.SpellAction
//
//interface SpellStrategy {
//    val action: SpellAction
//    val id get() = action.id
//    fun inZone(creature: Creature): Boolean
//    fun handle(time: Tsm, actions: ActionConsumer, map: GameMap): Boolean
//}
